### DISCUSSION TITLE
picking a language name and moving to the next phase

### DISCUSSION THREAD

**Discussion by mflatt** [racket/rhombus#510] [General]
In several of the bi-weekly meetings (#180), we discussed moving from the [Iterative Design phase](../blob/master/resources/plan.md#phase-2-iterative-design-years) to the [Integration phase](../blob/master/resources/plan.md#phase-3-integration-months-or-years). We have now covered all of the items from our todo list from [the most recent discussion](../discussions/180#discussioncomment-9037367), so it seems that we're ready.

The big step in that transition is picking a name for the language.

The plan's rationale for picking language name distinct from the project was that we would likely write things about the Rhombus project that turn out not to be true about the new language. As it happens, the language evolved in a relatively monotonic way, and we even published a paper using the project name where the paper's description of the language seems likely to stay accurate.

So, one perspective is that the name "Rhombus" is fine, and we should preserve the momentum that we have. We can pick "Rhombus" — new name, same as the old name.

Another perspective is that the name "Rhombus" is arbitrary (the first non-occupied "R" word I thought of while preparing RacketCon'19 slides the morning before speaking), and we can do better. To consider that route, we need alternatives.

**Reply in this discussion to brainstorm names.** Provide just one name in each reply, so others can offer informal vote-like reactions (by clicking the face icon in GitHub below the comment).

**Comment by mflatt** [comment]
Roar

This is one of few simple noise-related words I can find that does not seem to be taken.

**Reply by Metaxal** [reply]
I find this one a little difficult to say or even hear — I'm sure everyone is going to ask me to spell it.

**Comment by mflatt** [comment]
Rhombus

A placeholder for the current name, so we can collect reactions to the possibility of keeping it.

**Reply by Metaxal** [reply]
I like this name for its originality, its relation to mathematical principles (as in "we thought really hard while designing this new simple and clean language") and for not trying to be "cool" (what's cool one year is has-been the next one). My main worry though is that people who hear it for the first time might think it's too dedicated to geometry libraries and not a proper general programming language — though perhaps this worry is misguided?

**Reply by spdegabrielle** [reply]
Names have stickiness. I still see people use the first placeholder name.

**Reply by spdegabrielle** [reply]
@Metaxal i share your concerns

**Reply by samth** [reply]
I still like Rhombus more than any other suggestion. It departs from the Racket and Scheme tradition, it lends itself to logos and other graphics, it's memorable and unused, and it's already gained some mind share, including a published paper.

**Comment by Metaxal** [comment]
Shrubbery

**Reply by Lazerbeak12345** [reply]
The _Monty Python and the Holy Grail_ references would be inevitable with this one.

(not bad thing or good thing)

**Reply by spdegabrielle** [reply]
https://docs.racket-lang.org/shrubbery/language.html

**Reply by Metaxal** [reply]
@Lazerbeak12345 Then it may be too reminiscent of Python, which is probably not a good thing :/

**Comment by Metaxal** [comment]
Shrub

**Comment by spdegabrielle** [comment]
Rocket 🚀

Suggested by @jeapostrophe in #rhombus  https://discord.com/channels/571040468092321801/871953739982995547/893496711358586891

**Reply by lojic** [reply]
While I like Rocket, having the name differ by one letter from Racket, and a fairly subtle difference as far as letter differences go, would likely be too confusing.

**Reply by jeapostrophe** [reply]
I consider that to be a feature and I think that we (people in Racket) over-estimate the popularity and widespreadedness of Racket. Rocket is obviously the coolest name. ;) The kids will love it. Kids will not get "MaLex" or anything like that. If "we" are successful and Rhombus "works", then people won't even know or care about Racket in the same way that people who know about Racket today don't care, acknowledge, or think about "PLT Scheme" or "R5RS" in any way.

**Reply by lojic** [reply]
I accept the challenge of ensuring people continue to [know and care about Racket](https://xkcd.com/297/) :)

**Reply by spdegabrielle** [reply]
I like that it comes with emoji

🚀🚀🚀🚀🚀🚀🚀🚀

**Reply by Lazerbeak12345** [reply]
The IPA is also similar:

Racket -> [/ˈrækɪt/](http://ipa-reader.xyz/?text=%CB%88r%C3%A6k%C9%AAt&voice=Joanna)

Rocket -> [/ˈrɑkət/](http://ipa-reader.xyz/?text=%CB%88r%C9%91k%C9%99t&voice=Joanna) (American) or [/ˈrɒkɪt/](http://ipa-reader.xyz/?text=%CB%88r%C9%92k%C9%AAt&voice=Joanna) (British)

That is to say, if said aloud, it _could_ be hard to understand the distinction (especially with the British way of pronouncing Rocket)

**Reply by Metaxal** [reply]
To my ears, this name is trying too hard to be "cool" — leading to the opposite effect. Though perhaps 'Roquette' as in the salad? :)

**Reply by spdegabrielle** [reply]
Cool is good.
I’m glad you have resisted the urge to suggest the cheese

**Reply by otherjoel** [reply]
Rocket suggests an emphasis on speed though, which is most definitely not the case here

**Reply by jeapostrophe** [reply]
I think it suggests inter-planetary human expansion, defying boundaries, doing amazing things, being really cool, having impressive explosions, and Elton John, which I think are all extremely on-brand.

**Reply by usaoc** [reply]
I don’t like how this name could be associated with the infamous “blazing fast” Rust meme, especially with the emoji :P

**Reply by spdegabrielle** [reply]
I don’t know the rust meme.
Speed is not a focus because the current implementation is backed by a modern compiler that generates pretty fast native code.

**Reply by mflatt** [reply]
Not exactly a language, but a Rust Web framework uses this name: https://rocket.rs/guide/v0.5/introduction/

**Reply by spdegabrielle** [reply]
Want to give this another bump. Rockets are hot right now and will be for the foreseeable future. This is a great name.

**Comment by Metaxal** [comment]
MaLex

for "lexical" macros

**Comment by Metaxal** [comment]
MaStic

Macros with static information

**Comment by spdegabrielle** [comment]


**Reply by sorawee** [reply]
Just want to note one potential downside of this, which is that it could be super confusing to talk about the language in conversation. Which "Racket" are we talking about.

**Reply by Metaxal** [reply]
I suppose an English-native person could hear the difference, but a French one certainly wouldn't.

**Reply by spdegabrielle** [reply]
Good point 🤣

**Comment by sorawee** [comment]
Rogue (continuing the theme of illegal activities, starting with "R")

**Reply by rfindler** [reply]
I always picked the rogue character when I was a kid playing video games. So that'll appeal to a certain demographic 🤣

**Reply by benknoble** [reply]
There's also an opportunity to design a fun black and white logo (after the X-men character).

**Reply by mflatt** [reply]
Existing language with this name: https://github.com/brombres/Rogue

**Reply by spdegabrielle** [reply]
We don’t want to have the rust problem.  They share a name with a popular video game.

'rogue programming' ! is that programming in rogue-lang? or programming a rogue-like?

rogue-dev etc.

I'd also suggest the community is of comparable size.

**Reply by spdegabrielle** [reply]
I don't like the illegal activities aspect. Not for a professional tool.

https://www.collinsdictionary.com/sentences/english/rogue

**Reply by benknoble** [reply]
Although also possibly negative, "roguish"/"rogueish" can mean "mischievous" or "playful." As someone else pointed out, the Gambit synergy is fun, too.

I appreciate the professionalism concern, but I already usually have to explain what Racket is (to say nothing of absurd names like Terraform—infrastructure management has nothing to do with terraforming?—or even niche things like Dhall). Naming things is hard. To non-programmers, we already spew nonsense and gibberish.

Rogue not being _too_ serious is, for me, a feature rather than a bug.

**Reply by spdegabrielle** [reply]
It absolutely is a name with negative connotations https://www.collinsdictionary.com/dictionary/english/rogue#rogue__1

Check the sentences section for examples that include ‘rogue code’

**Comment by Metaxal** [comment]
Rupture

**Reply by jbclements** [reply]
First one to make me literally LoL. Not ... that that's probably a good thing. FWIW, I voted against the change from Scheme to Racket, so I'm already firmly on the wrong side of history there.

**Comment by lojic** [comment]
Raket

Racket without the paren (c). I agree with the pronunciation comment about Rhacket above, so we would have to pronounce it as "raked".

**Reply by Lazerbeak12345** [reply]
Something like [/reɪkt/](http://ipa-reader.xyz/?text=re%C9%AAkt&voice=Joanna)?

**Comment by spdegabrielle** [comment]


**Reply by spdegabrielle** [reply]
This was meant to be a reference to the Broca area of the brain linked to speech production. Not - according to GitHub translation - ‘drill bit’ in Spanish.

**Comment by soegaard** [comment]
Chameleon

**Reply by spdegabrielle** [reply]
I really like this one. I feel it expresses the capacity for change that macros on a bring

**Reply by Lazerbeak12345** [reply]
I at one point was making a competitor to racket by the same name - however I renamed it. It also has never become popular, because I discovered racket.

I'm not worried about the use of this name - but I don't remember why I changed _my_ project's name away from that. Perhaps there's already a name conflict?

**Reply by soegaard** [reply]
There is a language named [Caméléon](https://www.shinoe.org/cameleon/features.html).

**Reply by spdegabrielle** [reply]
I like it but I think it is too long and a good proportion of devs will need a spellchecker to spell it successfully

**Reply by jbclements** [reply]
Hmm... sounds like OCaml-eon

**Comment by ashton314** [comment]
Rivet

**Comment by AlexKnauth** [comment]
Shriek: Loud noise like Racket, starting with "Sh" like Shrubbery and Shplait

**Comment by Metaxal** [comment]
Whenever I'm thinking of a name, I'm thinking of someone asking me "What programming language do you use?" or "What programming language is this coded with?" Then I say "I use <name>" or "It's written in <name>", and try to get a feel about how my interlocutor would feel about my answer.

Because this is usually the situation in which  my interlocutor first hears about this language, and first impressions matter.

**Reply by sorawee** [reply]
Another thing is how long the name is. `import sixteencharacter/list` is a pain to type.

**Reply by spdegabrielle** [reply]
Boosting ‘first impressions matter’ because I feel rogue has too many negative connotations.

**Comment by bzyeung** [comment]
Quisling

**Reply by ashton314** [reply]
In this vein, "renegade" might be nice too. ~~I think I like "quisling" better.~~

**Reply by soegaard** [reply]
It's a bad idea to use the name of a Nazi collaborationist.

**Reply by ashton314** [reply]
Ah, I see the etymology now. Yeah, Quisling doesn't have a good connotation.

**Reply by sorawee** [reply]
Let me post "renegade" explicitly then.

**Reply by bzyeung** [reply]
I was inspired by the (S)cheme -> (R)acket -> (Q) pattern, plus Q-words being inherently distinctive.  If someone else finds it cute (albeit alphabet-specific), I think there are virtues to that underlying concept with an alternative word, even though I now agree that specific original word isn't a good choice.

**Comment by sorawee** [comment]
Renegade (continuing the theme of illegal activities, starting with "R")

**Comment by samdphillips** [comment]
Rhyme

a language that is like another language (Racket) but is not that language.

**Reply by spdegabrielle** [reply]
I love it

**Reply by LiberalArtist** [reply]
This also has a connection to sound, like "Racket" and "Shriek" (above)

**Reply by mflatt** [reply]
I like Rhyme, but note there are at least 3 languages with this name already.

**Reply by sorawee** [reply]
- https://github.com/rhyme-lang
- https://quantumzeitgeist.com/rhyme-the-high-level-quantum-programming-language/

What's the third one?

**Reply by mflatt** [reply]
https://github.com/rhyme-language/rhyme

**Comment by Metaxal** [comment]
Realm

**Reply by spdegabrielle** [reply]
This is good. Evocative. Short is a positive too.

**Reply by jackfirth** [reply]
This would get confusing given that there's a concept of "realms" in the racket docs already, and that was a feature added to support rhombus.

**Reply by sorawee** [reply]
Yeah, though I think these "lesser" names can be changed if Rhombus really wants to take their names, in the same way "Rumble" (that I proposed in Discord) can be changed. Rumble is an internal name though (0 results in https://docs.racket-lang.org/search/index.html?q=rumble), so it's definitely easier to take its name compared to Realm (5 definitions in https://docs.racket-lang.org/search/index.html?q=realm)

**Comment by spdegabrielle** [comment]
Anonymous suggestion that I like:

> I'm tempted to propose "Style", because who doesn't like programming with Style?

**Comment by samdphillips** [comment]
Rube

Like Rube Goldberg.

**Comment by spdegabrielle** [comment]
Wernicke after [Carl Wernicke](https://en.m.wikipedia.org/wiki/Carl_Wernicke)

**Comment by spdegabrielle** [comment]
Liskov

**Reply by spdegabrielle** [reply]
Turing award winner for contributions to PL and CS!

**Comment by spdegabrielle** [comment]
Bunny 🐰

**Reply by spdegabrielle** [reply]
This is just silly please ignore

**Comment by spdegabrielle** [comment]
***Apollo***

Searching ‘Apollo language’ does returns a English language school in Dublin - unlikely to be confused with a language.
 
Searching ‘apollo programming language’ gets results about the Apollo Guidance Computer (AGC) - cool but I think unlikely to cause confusion. I like that the name has a strong associations with Saturn V, Moon, etc.

**Reply by spdegabrielle** [reply]
And a nice emoji 🏹

**Comment by Lazerbeak12345** [comment]
Rock Programming Language :rock: (or perhaps :guitar: ??? since it's loud and all that?)

**Comment by samdphillips** [comment]
Rebus

There is at least a C# library with the same name.

**Reply by spdegabrielle** [reply]
And a Scottish TV detective

**Comment by benknoble** [comment]
Raclette

Inspired by both rocquette the salad, Chez's French, and being Racket's not-so-little-anymore sibling.

**Comment by soegaard** [comment]
Actually, by now I kind of like the name Rhombus.

**Comment by bzyeung** [comment]
Yawp

First of a bunch of brainstorms that came from reading the ideas and comments that have come so far.

**Reply by spdegabrielle** [reply]
Narp 
Edit: it felt like a ‘hot fuzz’(film) reference?

**Reply by bzyeung** [reply]
It means "harsh cry" or "raucous noise."   I saw Hot Fuzz, but don't recall the scene you're referencing. :blush: I was introduced to it via Walt Whitman.  The word is obscure for sure, but I don't know that popularity of the word itself matters for a language name.

**Comment by bzyeung** [comment]
Charade

**Comment by bzyeung** [comment]
Quip

**Comment by bzyeung** [comment]
Paddle

**Comment by bzyeung** [comment]
Bluff

**Comment by spdegabrielle** [comment]
Something to consider is the distinction between the language, the language implementation and the platform - using Racket for both has made communication about the project difficult and confusing.

**Reply by sorawee** [reply]
Can you show an example where "using Racket for both has made communication about the project difficult and confusing". I don't recall seeing people getting confused by this.

**Reply by spdegabrielle** [reply]
The ‘compiled/interpreted’ question is a classic - I've had people assume racket is interpreted because our website doesn’t  mention the primary implementation includes a modern native code compiler. (I know that being compiled or interpreted is not a property of the language, but it is undeniable that is affects choice)

I think it is sensible that different things have different names. Precedents include Java/jvm, c#/CLR, Elixir/BEAN.

**Reply by Lazerbeak12345** [reply]
I agree. I've tried to explain that things I've made are using racket, but can be compiled, and it wasn't easy.

**Comment by spdegabrielle** [comment]
I’d really like a name that sounds professional enough that I won’t feel silly suggesting it in a professional context.

**Comment by spdegabrielle** [comment]
Modern macros provide powerful, flexible, safe and accessible metaprogramming capability on modern compiler infrastructure - but not necessarily the Chez compiler.

 I think this means the naming needs to accommodate coexisting with many more languages on the platform (both embedded and external).

**Comment by eszpakowski** [comment]
Thicket

**Comment by spdegabrielle** [comment]
Arrow

**Comment by spdegabrielle** [comment]
Atoms 
because things are made of atoms

**Comment by eszpakowski** [comment]
Chant

**Comment by sorawee** [comment]
Argot.

Was going to suggest Slang, but it's already taken: https://github.com/slang-lang/slang. Thesaurus suggests Argot as an alternative.

Pidgin is also a suggestion, but it's already taken too: https://iss.oden.utexas.edu/?p=projects/pidgin

**Reply by spdegabrielle** [reply]
I like that

**Reply by spdegabrielle** [reply]
Also ‘Polari’ used by Punch and Judy performers

**Reply by benknoble** [reply]
I would avoid pidgin, which has often been used to classify languages as non-languages in order to suppress cultures.

**Comment by sorawee** [comment]
Macaron: from https://en.wikipedia.org/wiki/Macaronic_language. Note that "It is generally derogatory and used when the mixing of languages has a humorous or satirical intent or effect but is sometimes applied to more serious mixed-language literature.", though I don't know if that's still the case in modern usage.

Also close enough to macro.

Also confection.

**Reply by soegaard** [reply]
I like it! The logo is easy. I have only positive connotations wrt macarons.
And it it includes macros.

**Reply by soegaard** [reply]
Another reason to like `macaron`: There are "adjacent" names for projects.

Like in the Java-world where projects often have coffee related names.

**Reply by mflatt** [reply]
Not exactly languages, but two projects using this name: a Go web framework https://go-macaron.com/ and an analysis tool https://oracle.github.io/macaron/. I also find libraries like https://github.com/ecordell/pymacaroons that seem to implement a variant of cookies in the web-browser sense.

**Comment by jbclements** [comment]
Howl

Continues from Yawp and Racket, Ginsburg reference lands very vaguely near steal your face. (stretching further:) contains owl, mascot of Rice? Makes me think of Howland Owl, beloved Walt Kelly Pogo character? Leads toward Haskell/Howl collab called Thurston, rich buffoon from Gilligan's island?

I'm also taken by other un-cool noise-related names: weep, blather, din

Okay, sorry, back to work

**Reply by spdegabrielle** [reply]
An owl would be a great mascot. Is there a language called ‘Hoot’ yet?

**Comment by olopierpa** [comment]
A rebus is an intentionally obfuscated way of saying something. Not a good
name for a programming language.

**Reply by ashton314** [reply]
Rebuses are meant to be fun puzzles. I think it's a great name!

**Comment by Metaxal** [comment]
Rebound

**Comment by Metaxal** [comment]
Rebind

To give a new meaning to something

**Comment by eszpakowski** [comment]
Gist.

> noun. the choicest or most essential or most vital part of some idea or experience.

**Comment by ashton314** [comment]
Ni (pronounced "nee")

> You must bring us… a shrubbery! A nice one. And not too expensive…

**Reply by johannesrld** [reply]
this was also proposed on lobste.rs, I love it lol.

**Comment by sorawee** [comment]
Rumble

**Comment by mflatt** [comment]
From the June 27 meeting: The leading contenders so far are Rogue and Rhombus. We also liked Rumble, and so Sorawee added that one to the discussion.

Rogue had the most support. One problem is that a language already uses that name, but most felt that we could use it, anyway.

**Reply by LiberalArtist** [reply]
For anyone else catching up, here are the threads on [Rogue](https://github.com/racket/rhombus-prototype/discussions/510#discussioncomment-9775083) and [Rhombus](https://github.com/racket/rhombus-prototype/discussions/510#discussioncomment-9774468).

**Comment by spdegabrielle** [comment]
'Clay' or something similar for its plasticity

**Comment by LiberalArtist** [comment]
Quest

It fits the alphabetic `S`cheme -> `R`acket -> `Q`* pattern @bzyeung identified in https://github.com/racket/rhombus-prototype/discussions/510#discussioncomment-9777466, and it also fits thematically with Planner -> Conniver -> Scheme[r] -> Racket in the sense that you might have a quest, scheme, or plan to achieve some goal.

Conflicts: The closest seems to be a [text-adventure game engine](https://github.com/textadventures/quest) that uses a [scripting language called ASL](https://docs.textadventures.co.uk/quest/quest_code.html) which is described in [some places](https://github.com/textadventures/quest/blob/63f159054e088a40b3b22ed1954392c4d3cb9974/README.md?plain=1#L7) as "the Quest game language". There are also [VR headsets](https://www.meta.com/quest/), an [English teaching program in Toronto](https://studyquest.net/), a [translation service](https://languagequest.net/), and an [MIT research program](https://quest.mit.edu/about/vision-statement).

**Reply by spdegabrielle** [reply]
It’s nice and short, lacks negative connotations, and is easy to spell!

**Reply by sorawee** [reply]
I like the name as well, but want to note 
- https://github.com/sampersand/quest is an actual programming language, though it appears there are not a lot of users, and there's no active development.
- https://codingquest.io/ is a relatively well-known programming challenge site for high school students.

**Comment by mflatt** [comment]
Thanks, everyone, for brainstorming names and contributing to the discussion!

I'm not seeing consensus on a any of the names, yet, but I think that a consensus on continuing to use the name Rhombus is within reach.

Many other names with a lot of interest are already used in some relevant way: Rogue (video game known well enough that "Rougue-like" is a recognizable adjective), Rocket (web framework with an order of magnitude more GitHub stars than Racket), Rhyme (multiple languages), Macaron (multiple frameworks and tools), Quest (Meta), and even Rumble (video-sharing site). Opinions vary on how much these conflicts matter, but after thinking about it a while, picking an unoccupied name has become a primary concern for me. There are also other good objections to some of these names, such as having a nefarious connotation.

I think the main objection to the name Rhombus is that it doesn't have any intrinsic attraction or clever meaning, and we didn't want to miss an opportunity to pick a better name. That was certainly my motivation for driving this thread. But I also like the name Rhombus, and @samth has [summarized the good points well](#discussioncomment-9920222).

I've reached out to people who registered downvotes for the name Rhombus, and I believe they would happy enough to keep the name after all — now that we've done the due diligence of exploring other possibilities.

We'll take this up one more time at this week's meeting (#180).

**Reply by sorawee** [reply]
I think Rhombus is great. No objection from me.

But just to nitpick, for the "unoccupied name" concern, I think Rhombus the security camera system is probably as "occupied" as Rumble the video sharing site or Rogue the game, if not more. It's more relevant to programming, with [API](https://apidocs.rhombussystems.com/docs/introduction) that people can call. Microsoft in fact even has a [tutorial](https://learn.microsoft.com/en-us/entra/identity/saas-apps/rhombus-systems-provisioning-tutorial) on how to integrate Rhombus with their product.

**Reply by mflatt** [reply]
Oh, thank you for that pointer. As many times as I searched "rhombus" and "rhombus language", I never managed to notice that one. But I see now that I just missed it in the list of search results.

**Reply by otherjoel** [reply]
IMHO, the sheer current-day ubiquity of “rogue-like” actually lifts “rogue” away from any close association with the 44-year-old video game. Though no stranger to video games from all eras, I had myself completely forgotten that Rogue was an actual game. But I also think “Pong” and “Quake” would make great names for a new language, so take that for what it’s worth.

**Reply by rfindler** [reply]
I thought the concern about name "Rogue" being already occupied stemmed from this: https://github.com/brombres/Rogue

**Comment by mflatt** [comment]
In the meeting today, we agreed on **Rhombus** as the permanent name!

Next steps, expected to happen next week:
* This `rhombus-prototype` repo will be renamed `rhombus`.
* The renamed repo will be reorganized into multiple packages, such as `rhombus-lib`, `rhombus-doc`, `rhombus`, `rhombus-pict-lib`, etc. (That is, we won't create separate repos to hold different Rhombus packages, for now.)
* For near-term compatibility, we'll adjust the `rhombus-prototype` package registration so that it refers to a package in the reorganized repo, and that package will depend on a suitable set of the other packages.

**Reply by AlexKnauth** [reply]
Can we avoid splitting out `rhombus-doc`, and just put the documentation in the main `rhombus` package?

**Reply by rfindler** [reply]
> Can we avoid splitting out `rhombus-doc`, and just put the documentation in the main `rhombus` package?

I feel like there is some value in the ability to depend only on the implementation of some package to avoid pulling in tests and docs, but I'm not sure there is much value in depending on test and docs separately, so maybe a good idea is to have `rhombus-lib` and `rhombus-dev`, where `-dev` means both tests and docs (and CI support? .. and other stuff?). That also seems like it is going in the direction of existing conventions in OS-level package managers, kind of.

**Reply by benknoble** [reply]
I've been thinking about this some, too, for my own projects, and I think I'm happy with `project` and `project-lib`, where the former implies the latter _and_ contains docs, tests, etc. The lib package may have tests, too, depending on how split they are from the modules they test.

Qi has a `qi-sdk` package for what I think Robby might want in `-dev`, such as CI support and "other stuff."

**Reply by AlexKnauth** [reply]
The one with the tests and docs could just be the `rhombus` package, I don't see why a `-dev` suffix would help.

**Reply by LiberalArtist** [reply]
I've also been tentatively trying this two-package structure (`foo` and `foo-lib`, rather than four with `foo`, `foo-lib`, `foo-doc`, and `foo-test`).

**Reply by mflatt** [reply]
I'll go with `rhombus-lib` and `rhombus`.

I thought I might remember some reason for the choice, but the only motivation I remember for a more fine-grained split was to separate tests from documentation, which I think hasn't turned out to matter.

**Reply by LiberalArtist** [reply]
> I thought I might remember some reason for the choice, but the only motivation I remember for a more fine-grained split was to separate tests from documentation, which I think hasn't turned out to matter.

I can imagine it might matter for some specific package with very extensive tests or significant test dependencies, but the big dependency concern seems to be the large cycle involving `racket-doc`.

**Reply by mflatt** [reply]
Here's a tentative package split: https://github.com/mflatt/rhombus-prototype/tree/split

Every directory other than "resources" and "design" will be registered as a package. Then, to get a repo checkout with everything registered as a package, you'll be able to use `raco pkg install --clone path/to/rhombus rhombus-main-distribution`.
